src/Features/Core/Portable/IntroduceVariable/AbstractIntroduceParameterService.cs

```diff
@@ -0,0 +1,951 @@
+            IMethodSymbol methodSymbol, SyntaxNode containingMethod,
+            bool allOccurrences, string parameterName, TExpressionSyntax expression, bool trampoline, bool overload,
+            CancellationToken cancellationToken)
+        {
+            var generator = SyntaxGenerator.GetGenerator(currentDocument);
+            var semanticFacts = currentDocument.GetRequiredLanguageService<ISemanticFactsService>();
+            var invocationSemanticModel = await currentDocument.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+
+            var root = await currentDocument.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var editor = new SyntaxEditor(root, generator);
+            var syntaxFacts = currentDocument.GetRequiredLanguageService<ISyntaxFactsService>();
+            var insertionIndex = GetInsertionIndex(compilation, methodSymbol, syntaxFacts, containingMethod);
+
+            // Creating a new method name by concatenating the parameter name that has been camelcased.
+            var newMethodIdentifier = "Get" + parameterName.First().ToString().ToUpper() + parameterName[1..];
+
+            var validParameters = methodSymbol.Parameters.AsEnumerable().SelectMany(parameter => mappingDictionary.Values.Where(param => parameter.Equals(param))).ToImmutableArray();
+            if (trampoline)
+            {
+                foreach (var invocation in invocations)
+                {
+                    var argumentListSyntax = syntaxFacts.GetArgumentListOfInvocationExpression(invocation);
+                    editor.ReplaceNode(argumentListSyntax, (currentArgumentListSyntax, _) =>
+                    {
+                        return GenerateNewArgumentListSyntaxForTrampoline(syntaxFacts, semanticFacts, invocationSemanticModel, generator, currentArgumentListSyntax,
+                            argumentListSyntax, validParameters, newMethodIdentifier, insertionIndex, cancellationToken);
+                    });
+                }
+            }
+
+            // If you are at the original document, then also introduce the new method and introduce the parameter.
+            if (currentDocument.Id == originalDocument.Id)
+            {
+                if (trampoline)
+                {
+                    var newMethodNode = await ExtractMethodAsync(originalDocument, expression, methodSymbol, validParameters, newMethodIdentifier,
+                        generator, cancellationToken).ConfigureAwait(false);
+                    editor.InsertBefore(containingMethod, newMethodNode);
+                }
+                else if (overload)
+                {
+                    var newMethodNode = await GenerateNewMethodOverloadAsync(originalDocument, expression, methodSymbol, insertionIndex,
+                        generator, cancellationToken).ConfigureAwait(false);
+                    editor.InsertBefore(containingMethod, newMethodNode);
```

Comment:
which method has fewer parameters and calls into the original?  that one should go first.  if that's the case, this code is fine.  otherwise, reverse this.

Reply:
was this resolved?